src/modelio.cs

```diff
@@ -90,6 +90,28 @@ interface MDLAsset : NSCopying
 		[Export ("canExportFileExtension:")]
 		bool CanExportFileExtension (string extension);
 
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("components", ArgumentSemantic.Copy)]
+		IMDLComponent[] Components { get; }
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("setComponent:forProtocol:")]
+		void SetComponent (IMDLComponent component, Protocol protocol);
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("componentConformingToProtocol:")]
+		[return: NullAllowed]
+		IMDLComponent IsComponentConforming (Protocol protocol);
+
+		[Internal]
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		IMDLComponent ObjectForKeyedSubscript (Protocol key);
+
+		[Internal]
+		[Export ("setObject:forKeyedSubscript:")]
+		void SetObject ([NullAllowed] IMDLComponent obj, Protocol key);
```

Comment:
Yes so `objectForKeyedSubscript:`'s description is:

> Allows shorthand [key] syntax for `componentConformingToProtocol:`

@spouliot correct me if I'm wrong but this is basically the indexer I implemented initially (`public IMDLComponent this [Protocol key]` which was sentenced to die because there's already one).